src/design_notes/autoref_ops.md

```diff
@@ -0,0 +1,29 @@
+# Autoref/Autoderef in operators
+
+Rust permits overriding most of the operators (e.g., `+`, `-`, `+=`). In part
+due to `Copy` types not auto-dereferencing, it is common to have `T + &T`
+or `&T + &T`, with potentially many levels of indirection on either side of the
+operator.
+
+There is desire in general to avoid needing to both add impls for referenced
+versions of types because they:
+
+- bloat documentation
+- are never quite sufficient (always more references are possible)
+- can cause inference regressions, as the compiler cannot in general know that
+  `&T + &T` is equivalent to `u64 + u64`.
+
```

Comment:
I think this accurately captures the primary benefits of this: the compiler could avoid inference failures by considering the reference and non-reference possibilities without them being completely separate impls.

It might be worth mentioning something here about other tradeoffs. And in particular, we should consider whether there are circumstances where we might *not* want this behavior for a type, and whether people should be able to opt into or out of this behavior for a type.